src/Integration/Binding/CSharpVBBindingConfigProvider.cs

```diff
@@ -112,14 +114,40 @@ private async Task<IBindingConfig> DoGetConfigurationAsync(SonarQubeQualityProfi
             // Finally, fetch the remaining data needed to build the ruleset
             var inactiveRules = await FetchSupportedRulesAsync(false, qualityProfile.Key, cancellationToken);
 
+            var ruleset = GetRulesetFile(qualityProfile, language, bindingConfiguration, activeRules, inactiveRules, sonarProperties);
+            var additionalFile = GetAdditionalFile(language, bindingConfiguration, activeRules, sonarProperties);
+
+            return new CSharpVBBindingConfig(ruleset, additionalFile);
+        }
+
+        private FilePathAndContent<VsRuleset> GetRulesetFile(SonarQubeQualityProfile qualityProfile, Language language, BindingConfiguration bindingConfiguration, IEnumerable<SonarQubeRule> activeRules, IEnumerable<SonarQubeRule> inactiveRules, Dictionary<string, string> sonarProperties)
+        {
             var coreRuleset = CreateRuleset(qualityProfile, language, bindingConfiguration, activeRules.Union(inactiveRules), sonarProperties);
 
             var ruleSetFilePath = solutionBindingFilePathGenerator.Generate(
                 bindingConfiguration.BindingConfigDirectory,
                 bindingConfiguration.Project.ProjectKey,
                 language.FileSuffixAndExtension);
 
-            return new CSharpVBBindingConfig(ToVsRuleset(coreRuleset), ruleSetFilePath);
+            var ruleset = new FilePathAndContent<VsRuleset>(ruleSetFilePath, ToVsRuleset(coreRuleset));
+
+            return ruleset;
+        }
+
+        private FilePathAndContent<SonarLintConfiguration> GetAdditionalFile(Language language, BindingConfiguration bindingConfiguration, IEnumerable<SonarQubeRule> activeRules, Dictionary<string, string> sonarProperties)
+        {
+            var additionalFilePathDirectory = solutionBindingFilePathGenerator.Generate(
+                bindingConfiguration.BindingConfigDirectory,
+                bindingConfiguration.Project.ProjectKey,
+                string.Empty);
+
+            var additionalFilePath = Path.Combine(additionalFilePathDirectory, language.Id, "SonarLint.xml");
```

Comment:
I'm not a fan of this split: the reason for this is that solutionBindingFilePathGenerator uses PathHelper.EscapeFileName, which replaces \ to _, and we need the file name to be precisely SonarLint.xml, and not csharp_sonarlint.xml.

Reply:
As previously discussed, I think the language-specific config providers should be entirely responsible for calculating their own file paths. They should be handed the base directory to use, and left to work everything else out for themselves.
Here, the base directory is in `bindingConfiguration`. You don't actually need the `solutionBindingFilePathGenerator` at all.
Just build the whole path yourself. If you need call the PathHelper directly, do so. Or add an extension method to `BindingConfiguration` that returns the `EscapedProjectName` in a format that can be used in a  filename.